File: virtual-world-cpp/VirtualWorld/Plants/Nightshade.cpp
#include "Nightshade.h"

Nightshade::Nightshade(int x, int y, World *world) : Plant(x, y, 99, 0, nightshadeCode, world) {}

Plant *Nightshade::Clone(int x, int y, World *world) {
    return new Nightshade(x, y, world);
}

std::string Nightshade::GetName() {
    return "Nightshade";
}

bool Nightshade::AttackPaired(Organism *attacker) {
    attacker->Kill();
    this->Kill();
    this->world->GetTile(getX(), getY())->SetOrganism(nullptr);
    this->world->GetTile(attacker->getX(), attacker->getY())->SetOrganism(nullptr);
    this->world->AddMessage(attacker->GetName() + " ate nightshade and died!");
    return true;
}

File: virtual-world-java/src/main/java/me/knasuta/virtualworld/simulation/plants/Nightshade.java
package me.knasuta.virtualworld.simulation.plants;

import me.knasuta.virtualworld.simulation.Organism;
import me.knasuta.virtualworld.simulation.Point;
import me.knasuta.virtualworld.simulation.World;

import java.awt.*;

public class Nightshade extends Plant {
    public Nightshade(Point location, World world) {
        super(location, 99, 0, world);
    }
    @Override
    public boolean AttackPaired(Organism other) {
        other.Kill();
        this.Kill();
        world.RemoveOrganism(other);
        world.RemoveOrganism(this);
        world.AddMessage(other.getName() + " ate nightshade and died");
        return true;
    }

    @Override
    public String getName() {
        return "Nightshade";
    }

    @Override
    public Organism Clone(Point location, World world) {
        return new Nightshade(location, world);
    }

    @Override
    public Color getColor() {
        return Color.BLACK;
    }
}
